src/dbnode/encoding/proto/README.md

```diff
@@ -0,0 +1,269 @@
+# Protobuf Encoding
+
+## Overview
+
+This package contains the encoder/decoder(iterator) for compressing streams of Protobuf messages matching a provided schema.
+All compression is performed in a streaming manner such that we update the encoded stream with each write; there is no internal buffering or batching during which we gather multiple writes before performing encoding.
+
+## Features
+
+1. Compress the timestamps of the ProtoBuf messages using [Gorilla-style delta of delta encoding](https://www.vldb.org/pvldb/vol8/p1816-teller.pdf).
+2. Compress streams of ProtoBuf messages matching a provided schema by using different forms of compression for each field based on its type.
+3. Support changing the schema of a ProtoBuf message mid-stream (pending).
+4. Absolutely no lossiness for any of the compression types.
+
+## Supported Syntax
+
+This package strives to support compressing any valid Protobuf messages as specified using [Protobuf syntax version 3](https://developers.google.com/protocol-buffers/docs/proto3), however, only the following features have been tested:
+
+1. [All scalar value types](https://developers.google.com/protocol-buffers/docs/proto3#scalar)
+2. Nested Messages
+3. Repeated fields
+4. Map fields
+5. Reserved fields
+
+We have explicitly not performed any testing of the following features and thus they are not supported:
+
+1. `Any` fields
+2. `Oneof` fields
+3. Options of any type
+4. Custom field types
+
+## Compression Techniques
+
+This package compresses the timestamps for the Protobuf messages using [Gorilla style delta of delta encoding](https://www.vldb.org/pvldb/vol8/p1816-teller.pdf).
+
+Additionally, we perform a different type of compression for each field in the Protobuf message based on its type so that we can use compression techniques that are highly efficient for that specific type.
+
+1. Float fields are compressed using [Gorilla style XOR compression](https://www.vldb.org/pvldb/vol8/p1816-teller.pdf).
+2. Integer fields are compressed using M3TSZ significant digit integer compression. We don't currently have any documentation on this compression format.
```

Comment:
Is this delta-of-delta?  I can't remember, hah.

Reply:
nope just one delta